src/webgl/shaders/filters/invert.frag

```diff
@@ -9,6 +9,11 @@ uniform sampler2D tex0;
 
 void main() {
   vec4 color = texture2D(tex0, vTexCoord);
-  vec3 invertedColor = 1.0 - color.rgb;
-  gl_FragColor = vec4(invertedColor, color.a);
+  vec3 invertedColor = vec3(1.0) - color.rgb; // Declare and calculate invertedColor
+  //checking if pixel is opaque
+  if (color.a == 1.0) { 
```

Comment:
I think this will still break for colors that are semitransparent, since they'll go through the `else` branch.

To explain a bit about what's happening, the colors here are premultiplied, meaning `color.r` is actually the original `r` value multiplied by `color.a`. 

So if we want to invert just the original colors but keep the alpha, maybe we need:
```glsl
vec3 origColor = color.rgb / color.a;
vec3 invertedColor = vec3(1.0) - origColor;
gl_FragColor = vec4(invertedColor * color.a, color.a);
```